hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/parsing/PopulateJMSMessageHeader.java

```diff
@@ -0,0 +1,36 @@
+/**
+ * 
+ */
+package ca.bc.gov.hlth.hnsecure.parsing;
+
+import org.apache.camel.Exchange;
+import org.apache.camel.Handler;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ca.bc.gov.hlth.hncommon.util.LoggingUtil;
+
+/**
+ * Add hexadecimal message ControlId with a prefix ID: to stop any encoding in MQRFH2
+ * Refer : https://www.ibm.com/docs/en/ibm-mq/7.5?topic=messages-mapping-jms-header-fields-send-publish
+ * @author anumeha.srivastava
+ *
+ */
+public class PopulateJMSMessageHeader {
+	private static final Logger logger = LoggerFactory.getLogger(PopulateJMSMessageHeader.class);
+
+	@Handler
+	public void populateJMSRequestHeader(Exchange exchange, String v2Message) {
+		final String methodName = LoggingUtil.getMethodName();
+		String msgControlId = V2MessageUtil.getMsgCnrtlId(v2Message);
+		String hexString = Util.convertStringToHex(msgControlId);
+		
+		exchange.getIn().setHeader("JMSMessageID", hexString);
+		exchange.getIn().setHeader("JMSCorrelationID", "ID:" + hexString);	
+						
+		logger.info("{} - Transaction Id : {}, JMS messageId is set to : {}, JMS correlationId is set to : {}  ", methodName, exchange.getExchangeId(),
+				hexString,hexString);
```

Comment:
Please add a space between arguments (hexString).
Also, the second hexString argument isn't technically correct since you're really trying to log correlationId. Probably best to store correlationId as a local variable and then use it in the header and log statement.